### PR TITLE
[7.67.x][JBPM-10125] Test for write raw to avoid escape chars in SOAP headers

### PR DESCRIPTION
**Pull request by gmunozfe**
Backported from bc2298b236027bc88daf5caa8a0e99d49a1163b0 and 8f2609d04c43877b8123c6f0fd86d9405c5ac652

**JIRA**: [JBPM-10125](https://issues.redhat.com/browse/JBPM-10125)

referenced Pull Requests:

- jbpm#2226(https://github.com/kiegroup/jbpm/pull/2226)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-test-web-service/src/main/java/com/jboss/soap/service/acmedemo/impl/HeaderInInterceptor.java [hunk 1]
```diff
@@ -38,26 +31,12 @@ public void handleMessage(SoapMessage message) throws Fault {
        logger.info("headerContent is {}", headerContent);
 
        //DumbEscapeHandler modifies chars over US-ASCII
-       if ("Espa&#241;a".equals(headerContent)) {
+       if ("Espa&#241;a".equals(headerContent) || "AT&T".equals(headerContent)) {
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-test-web-service/src/main/java/com/jboss/soap/service/acmedemo/impl/HeaderInInterceptor.java:34]: As far as I understood we write `<Coupon><Number><![CDATA[AT&T]]></Number></Coupon>` as `<Coupon><Number>AT&T</Number></Coupon>`, don't we? So why do we no longer need the XPath expression to extract the coupon number? Where does this extraction happen?

**Review by MarianMacik — APPROVED** [review]
Just a question out of curiosity. Since it is a backport, approving already.

File: kie-server-parent/kie-server-tests/kie-server-test-web-service/src/main/java/com/jboss/soap/service/acmedemo/impl/HeaderInInterceptor.java (hunk 1 above)
Comment (gmunozfe) [reply kie-server-parent/kie-server-tests/kie-server-test-web-service/src/main/java/com/jboss/soap/service/acmedemo/impl/HeaderInInterceptor.java:34]: `getTextContent` now returns only the text under the xml nodes (`AT&T`), while before it was returning the whole xml stringfied: `<Coupon><Number>AT&T</Number></Coupon>`

**Comment by gmunozfe** [comment]
@mareknovotny could you merge?

**Pull request merged**